cmd/tidb-server/main.go

```diff
@@ -677,14 +677,12 @@ func overrideConfig(cfg *config.Config, fset *flag.FlagSet) {
 	}
 
 	if actualFlags[nmTiDBServiceScope] {
-		scope, ok := distroleutil.ToTiDBServiceScope(*serviceScope)
-		if !ok {
-			err := fmt.Errorf("incorrect value: `%s`. %s options: %s",
-				*serviceScope,
-				nmTiDBServiceScope, `"", background`)
+		if servicescope.IsValidServiceScope(*serviceScope) {
+			cfg.Instance.TiDBServiceScope = *serviceScope
+		} else {
+			err = fmt.Errorf("the tidb-service-scope value '%s' is invalid. It must be 64 characters or fewer and consist only of letters (a-z, A-Z), numbers (0-9), hyphens (-), and underscores (_)", *serviceScope)
```

Comment:
When you limit value length, you should sync with @frank945946？and also need to change the doc to mention it.